apache/accumulo
# VolumeManager and Volume internals cleanup

* VolumeManager
  * remove getDefaultVolume, which was only used to create a temporary
    directory for creating tables with split points, and to resolve
    relative paths
  * replace "default volume" concept with the first available volume
    (this is what it was effectively doing anyway... but now it's
    implemented more cleanly)
  * simplified much of the volume management code in the CreateTable
    FaTE operation classes, where creation of temporary files for
    storing splits on a pre-split table was being handled
  * rename getFileSystem methods throughout that return VolumeManager to
    getVolumeManager, and stop passing it as an extra parameter if it is
    already available via a ServerContext object
* Volume
  * Remove unneeded overloaded Volume.prefixChild(Path)
  * Rename Volume.isValidPath(Path) to Volume.containsPath(Path) to
    improve readability
  * Update javadoc for containsPath to describe what it means for a Path
    to be contained
  * Update VolumeImplTest to verify behavior of containsPath
* VolumeImpl
  * Make protected fields private
  * Rename conf to hadoopConf to clarify which config it is
  * normalize base path by stripping out trailing slashes
  * rename helper method equivalentPaths(Path) to isAncestorPathOf(Path)
    for readability in containsPath(Path) implementation
  * add more strict argument checking for prefixChild(String)
  * fix bug in isAncestorPathOf(Path) that incorrectly concluded /a/path
    is an ancestor of /a/pa (/a is an ancestor, but /a/pa is not)
* VolumeImplTest
  * check normalization by testing with trailing slashes
  * add more test cases for isAncestorPathOf, including checking for
    "breakout" path terms, like ".."
  * add test cases for prefixChild and containsPath
* Tablet
  * Fix bug in Tablet that wasn't using the Path's FileSystem when
    deleting a file, but instead was using an arbitrary FileSystem
